OvmfPkg/PlatformPei/PlatformId.c

```diff
@@ -0,0 +1,660 @@
+ * does not exceed BufferEnd and that the string is NUL-terminated.
+ *
+ * @param[in] Buffer  The buffer to read from.
+ * @param[in] BufferEnd  The end of the buffer.
+ * @param[out] Start  The start of the string.
+ * @param[out] End  The byte after the NUL-terminator.
+ *
+ * @retval EFI_SUCCESS  The string was read successfully.
+ * @retval EFI_BUFFER_TOO_SMALL  The string's declared size was too large for
+ *                               the buffer.
+ * @retval EFI_INVALID_PARAMETER  The string was not NUL-terminated.
+ */
+STATIC
+EFI_STATUS
+ConsumeAsciiByteStr (
+  IN  UINT8  *Buffer,
+  IN  UINT8  *BufferEnd,
+  OUT UINT8  **Start,
+  OUT UINT8  **End
+  )
+{
+  EFI_STATUS  Status;
+  UINT8       EvtSize;
+
+  Status = SafeOffset (Buffer, BufferEnd, sizeof (EvtSize), Start);
+  if (EFI_ERROR (Status)) {
+    return Status;
+  }
+
+  EvtSize = *Buffer;
+  // The spec does not allow for a zero-length string since it must be NUL-
+  // terminated.
+  if (EvtSize == 0) {
+    DEBUG ((DEBUG_ERROR, DPREFIX "Expected a NUL-terminated string\n"));
+    return EFI_INVALID_PARAMETER;
+  }
+
+  Status = SafeOffset (*Start, BufferEnd, EvtSize, End);
+  if (EFI_ERROR (Status)) {
+    return Status;
+  }
+
+  if (*(*End - 1) != 0) {
+    DEBUG ((DEBUG_ERROR, DPREFIX "Expected a NUL-terminated string\n"));
+    return EFI_INVALID_PARAMETER;
+  }
+
+  return EFI_SUCCESS;
+}
+
+/**
+ * Reads a first 32-bit integer as a size, then determines if Buffer + 4 + size
+ * exceeds BufferEnd.
+ *
+ * @param[in] Buffer  The buffer to read from.
+ * @param[in] BufferEnd  The end of the buffer.
+ * @param[out] Start  The start of the string.
+ * @param[out] End  The byte after the last character of the string.
+ *
+ * @retval EFI_SUCCESS  The string was read successfully.
+ * @retval EFI_BUFFER_TOO_SMALL  The string's declared size was too large for
+ *                               the buffer.
+*/
+STATIC
+EFI_STATUS
+ConsumeUint32Str (
+  IN  UINT8  *Buffer,
+  IN  UINT8  *BufferEnd,
+  OUT UINT8  **Start,
+  OUT UINT8  **End
+  )
+{
+  EFI_STATUS  Status;
+  UINT32      EvtSize;
+
+  Status = ParseUint32 (Buffer, BufferEnd, &EvtSize, Start);
+  if (EFI_ERROR (Status)) {
+    return Status;
+  }
+
+  return SafeOffset (*Start, BufferEnd, EvtSize, End);
+}
+
+/**
+ * Validates that the locator is valid.
+ *
+ * PREREQUISITE: Locator <= LocatorEnd
+ *
+ * @param[in] Locator  The locator to validate.
+ * @param[in] LocatorEnd  The end of the locator.
+ * @param[in] LocatorType  The type of the locator.
+ *
+ * @retval EFI_SUCCESS  The locator is valid.
+ * @retval EFI_BUFFER_TOO_SMALL  The locator is too small.
+ * @retval EFI_INVALID_PARAMETER  The locator is invalid.
+**/
+STATIC
+EFI_STATUS
+ValidateLocator (
+  IN UINT8   *Locator,
+  IN UINT8   *LocatorEnd,
+  IN UINT32  LocatorType
+  )
+{
+  UINTN  Size;
+
+  // No validation done for non-UEFI variable locators.
+  if (LocatorType != LOCATOR_TYPE_UEFI_VARIABLE) {

```

Comment:
It seems you only want to support UEFI variable.

Then the func name should be `ValidateUefiVariableLocator()`